doc/copp/CoPP Config and Management.md

```diff
@@ -0,0 +1,111 @@
+# Introduction
+
+The scope of this document is to provide the requirements and a high-level design proposal for Sonic Control Plane Policing configuration and management.
+
+# Requirements
+
+The following are the high level requirements for CoPP/TRAP managment
+
+1. CoPP Tables shall be part of the Config DB. Currently this is only present in APP DB
+2. User shall be able to configure, modify CoPP table entries for various policing requirements
+3. System shall be backward compatible with old Config DB without any CoPP Tables
+4. A VS test must be added to verify the Config DB CoPP tables
+
+# Design Proposal
+
+## Current behaviour
+
+During SWSS start, the prebuilt [copp.json](https://github.com/Azure/sonic-swss/blob/201911/swssconfig/sample/00-copp.config.json) file is loaded as part of start script [swssconfig.sh](https://github.com/Azure/sonic-buildimage/blob/201911/dockers/docker-orchagent/swssconfig.sh) and written to APP DB. [CoppOrch](https://github.com/Azure/sonic-swss/blob/201911/orchagent/copporch.cpp) then translates it to Host trap/policer tables and programmed to SAI.
+
+## Proposed behaviour
+
+With the new proposal, the CoPP tables shall be loaded to Config DB instead of APP DB. The default CoPP json file shall be prebuilt to the image and loaded during initialization. Any Config DB entries present shall be configured overwriting the default CoPP tables. This also ensures backward compatibility. At a high-level, the following are the expected changes:
+
+### Schema Changes
+
+### Config DB
+
+```
+key = "COPP|name"
+name_list     = name | name,name_list
```

Comment:
Can we have a name generically defined instead of being a list of trap ids like in copp config file? For example like COPP|trap_group1. This would be helpful in maintaining the name consistent across upgrades and in scenarios where we add new trap ids to the group.